### ISSUE TITLE
debugger error in Centos

### ISSUE THREAD

**Issue by joniale** [de-jcup/eclipse-bash-editor#150]
Dear eclipse bash editor developers,

I have tried out your editor because of the debugger feature.

I am using Oxygen a3 eclipse under Centos 6.3
Our Centos version does not have the "x-terminal-emulator" but "gnome-terminal".
I tried to change the XTerminal command under Preference->Bash editor->Debug. 
Unfortunately, i change it, i applied/saved configuration and i pressed test it, but it did NOT work. No matter what i write under XTerminal command that the plugin will ignore that. :/

As a workaround for that i created a link from "x-terminal-emulator" to "gnome-terminal".

Now i am having something like the following command when i activate the meta information debug in the console:
**bash -c x-terminal-emulator  -e bash --login -c 'cd /tmp;./terminallaunchblablabla.sh -a 1 -b 2;_exit_status=$?;echo "blablabla"'**

Notice i removed some parts using blablabla :)

In the terminal, i see the following:
Option "**--login**" is no longer supported in this version of gnome-terminal; you might want to create a profile with the desired setting, and use the new '--profile' option
Failed to parse arguments: **Unknown option -c**
result:1
------------------
So now it is running the gnome-terminal thanks to the link but there are some problems with the options.

What am i doing wrong here?
Any help or hint would be appreciated.
Thanks

**Comment by de-jcup** [comment]
Hmm.. looking at
![screenshot](https://user-images.githubusercontent.com/11644753/53704410-bcdb3600-3e1c-11e9-911b-50e1e9925a5d.png)

I should provide a more customizable way.
The problem seems to be the option "--login". Because you have no option to customize your command this is currently not possible.

I will try to make this configurable in next release

**Comment by joniale** [comment]
Hi,
Thanks for the fast answer and nice to see that this will be included in the next release.
Do you have an estimation for the release date?.

And another remark, as i said in the ticket, the XTerminal command field (on debug preferences) is not working. I can write whatever i want that the tool will always take the value x-terminal-emulator -e.
Thanks again
Jonathan

**Comment by de-jcup** [comment]
There is no release plan, but I think 2.1.0 will come in the next 1-3 Weeks.

**Comment by de-jcup** [comment]
Please do me a favor:
`bash -c x-terminal-emulator -e bash --login -c 'cd /tmp;./terminallaunchblablabla.sh -a 1 -b 2;_exit_status=$?;echo "blablabla"'`
Does not work for you.

Please try out which command does the job correctly at centos (i got no installation) and give me feedback which command works.

It is necessary to have got a command where a bash is opened as "login shell" this is the reason for "--login" option on gnome terminal. Its necessary to have an option doing same way on your setup. These shells will be kept open.... so we got necessary output for debugging.

**Comment by joniale** [comment]
Hi,

i have only found a solution using the GUI of the terminals. Unfortunately, i have not found a way to change this login shell configuration from the command line.

In **gnome terminals** there is a default terminal configuration . Any user can change the default behavior of the gnome-terminal command by using the graphical menu of any open terminal.
1. Edit -> Profile Preferences.
2. Select the Title and Command tab.
3. Notice how the Run command as login shell checkbox is unchecked! Check it.
That's it. If you do this to the **Default profile**, or to whatever profile is configured to be used when making new terminals, you get a **login shell**.

If you do not want to use the **default profile**   you can create a profile for login terminals and then execute that profile using:
**gnome-terminal  --profile=PROFILE-NAME**          Use the given profile instead of the default profile
I think that is what is meant with the error message:

>  Option "--login" is no longer supported in this version of gnome-terminal; you might want to create a profile with the desired setting, and use the new '--profile' option

In other words, Gnome-terminal users of your plugin need an extra step which is to create a profile for login shells and configure it on the call 
I hope that helps. 

Note:
Strangely, if i execute in my gnome-terminal:
bash --login
That command works without error/problems and there is no such message as "Option "--login" is no longer supported ". However, the result is a bash **that is NOT a login shell**.

If that is not what you wanted, then please let me know.
Jonathan

**Comment by de-jcup** [comment]
@joniale I moved this issue to milestone 2.2.0 because I wanted to bring already implemented features to marketplace. But this will be the next issue I will start and so It will be inside v2.2.0

**Comment by de-jcup** [comment]
Summary from doc:

1. User can setup full command, we got at least two variables inside: 
  * ${BE_CMD_TITLE}
   * ${BE_CMD_CALL}
2. Generic appraoch will provide defaults for win and Linux/mac.
3. Old x-terminal preference will be deprecated, we use new preference entry for generic termianl 
   command. 
4. Near Test-Button (for executing a test script) will be also another button where user an
   check the combined output for test, so he/she can check it on console before calling in eclipse.

**Comment by joniale** [comment]
Hi,
Nice to hear that. When will we have a release to test it?
Best regards,
Jonathan

**Comment by de-jcup** [comment]
Now launch is full configurable. Works at least with linux:
![image](https://user-images.githubusercontent.com/11644753/57894914-3219ad00-7849-11e9-9bac-1a9bb57c0d29.png)

Defaults for Windows must be checked, also documentation, UI must be still improved. Also some unit tests must be fixed.

**Comment by de-jcup** [comment]
@joniale I think next release will be in 4-7 days.

**Comment by de-jcup** [comment]
@joniale just release 2.2.0 on marketplace - includes changes

**Comment by joniale** [comment]
Hi,
Unfortunately, i have some problems to make it work with my gnome-terminal.
I had to modify the line using:
gnome-terminal -e "bash -c \'cd /tmp;./terminallaunch5648577765558613195.sh -a 1 -b 2;_exit_status=$?;echo "Exit code=$_exit_status";if [ $_exit_status -ne 0 ]; then read -p "Unexpected exit code:$_exit_status , press enter to continue";fi\'"

Notice that i introduced the quotes before bash so that the gnome-terminal can take the bash command.

I receive the word **"OK"** when i run the debug mode over a script.
However, the script will not be executed.
Do you have any idea what it is happening?
Best regards,
Jonathan

**Comment by de-jcup** [comment]
Hello @joniale 

I installed gnome-terminal and tried out by myself. ( I am using now Gnome Terminal 3.28.1)

When I define `gnome-terminal -- bash --login -c '${BE_CMD_CALL}'` as `Terminal command` and keep others as default (use `Restore Defaults` button when too much changed before...")

Pressing `Test Terminal` button brings up then:

![image](https://user-images.githubusercontent.com/11644753/58058024-9305f580-7b67-11e9-87cf-f8c2005a05c0.png)


_(By the way x-terminal-emulator is a only perl-script which calls installed terminals in a generic way when installed, so maybe it would be much easier if you would just simply install x-terminal-emulator per `sudo apt install x-terminal-emulator` ... I just did some print instructions to inspect the execution from x-terminal-emulator to get things working directly with gome-terminal)_

**Comment by joniale** [comment]
Hi,

As i said, in order to make it work, i changed the Terminal command to this
`gnome-terminal -e "bash --login -c \'${BE_CMD_CALL}\'"`

Please notice that it is included a "-e" to execute the bash inside the terminal and quotes "" for the bash sub-command. The simple quote used in the bash command '' is escaped with "\".

Your command `gnome-terminal -- bash --login -c '${BE_CMD_CALL}'` has a space between the first "--" and the word bash. Therefore it is not working.
With my command it returns OK, but then it is closed very fast. That can only be seen in the terminal were you called eclipse.

Regarding the installation of the x-terminal-emulator, is something i can try.

**Comment by de-jcup** [comment]
"-- bash" is not a typo and it is working. Look at the picture above: I really used the gnome terminal call.

When you call gnome-terminal with option `-e`
you have following output:

> \# Option “-e” is deprecated and might be removed in a later version of gnome-terminal.\# 
> \# Use “-- ” to terminate the options and put the command line to execute after it.\# 

Also the given line is exactly what x-terminal-emulator does when it calls internally gnome terminal.

**Comment by joniale** [comment]
Hi,
OK i understand now. 
I have tried with your line and **yes!! it is working with gnome-terminal**.
Maybe you can include that line in your tutorial about the debugger (gnome-terminal).
Thanks a lot
Jonathan

**Comment by de-jcup** [comment]
Hello, I have already added the gnome-terminal description to WIKI (see https://github.com/de-jcup/eclipse-bash-editor/wiki/Debugger#gnome-terminal)
